resources/hooks/local/pre-push

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+remote="$1"
+
+while read local_ref local_sha remote_ref remote_sha
+do
+    # Get the branch name
+    local_ref=$(basename $local_ref)
+    
+    # Fetch the GIT diff and format it as command input:
+    DIFF=$(git diff --name-only "$remote"/"$local_ref"..HEAD --oneline)
+        
+    # Run GrumPHP
+    (cd "./" && printf "%s\n" "${DIFF}" | exec 'vendor/phpro/grumphp/bin/grumphp' 'git:pre-push')
```

Comment:
This will run grumphp multiple times on multiple commits in a push. 
You want to create ONE diff from ALL the changed commits and run grumphp only once with this diff.
You could also pass the parameters that you receive in this hook as parameters to the cli.
This way it is possible to add them to the context and use them in the tests. For example: You can create a task that does not allow you to push upstream.